backend/src/main/java/fk/prof/backend/http/LeaderHttpVerticle.java

```diff
@@ -70,14 +72,58 @@ private Router setupRouting() {
     HttpHelper.attachHandlersToRoute(router, HttpMethod.GET, apiPathForGetWork,
         BodyHandler.create().setBodyLimit(1024 * 100), this::handleGetWork);
 
-    //TODO: Rework this once policy store hack is removed
-    String apiPathForUpdatingPolicy = ApiPathConstants.LEADER_POST_POLICY + "/:appId/:clusterId/:procName";
-    HttpHelper.attachHandlersToRoute(router, HttpMethod.POST, apiPathForUpdatingPolicy,
-            BodyHandler.create().setBodyLimit(1024 * 100), this::handlePostPolicy);
+    HttpHelper.attachHandlersToRoute(router, HttpMethod.GET, ApiPathConstants.LEADER_GET_APPIDS, this::handleGetAppIds);
+    HttpHelper.attachHandlersToRoute(router, HttpMethod.GET, ApiPathConstants.LEADER_GET_CLUSTERIDS_GIVEN_APPID, this::handleGetClusterIds);
+    HttpHelper.attachHandlersToRoute(router, HttpMethod.GET, ApiPathConstants.LEADER_GET_PROCNAMES_GIVEN_APPID_CLUSTERID, this::handleGetProcNames);
+
+    HttpHelper.attachHandlersToRoute(router, HttpMethod.GET, ApiPathConstants.LEADER_GET_POLICY_GIVEN_APPID_CLUSTERID_PROCNAME, this::handleGetPolicy);
+    HttpHelper.attachHandlersToRoute(router, HttpMethod.PUT, ApiPathConstants.LEADER_PUT_POLICY_GIVEN_APPID_CLUSTERID_PROCNAME,
+        BodyHandler.create().setBodyLimit(1024), this::handleUpdatePolicy);
+    HttpHelper.attachHandlersToRoute(router, HttpMethod.POST, ApiPathConstants.LEADER_POST_POLICY_GIVEN_APPID_CLUSTERID_PROCNAME,
+        BodyHandler.create().setBodyLimit(1024), this::handleCreatePolicy);
 
     return router;
   }
 
+  private void handleGetAppIds(RoutingContext context) {
+    try {
+      final String prefix = context.request().getParam("prefix");
+      context.response().end(Json.encode(policyStore.getAppIds(prefix)));
+    } catch (Exception ex) {
+      HttpFailure httpFailure = HttpFailure.failure(ex);
+      HttpHelper.handleFailure(context, httpFailure);
+    }
+  }
+
+  private void handleGetClusterIds(RoutingContext context) {
+    try {
+      final String appId = context.request().getParam("appId");
+      String prefix = context.request().getParam("prefix");
+      if (prefix == null) {
```

Comment:
Will update the store and its doc to do no filter if null is passed for prefix, but for other it will still throw NPE